lib/puppet/defaults.rb

```diff
@@ -362,6 +362,11 @@ def self.default_diffargs
       :desc    => "The hiera configuration file. Puppet only reads this file on startup, so you must restart the puppet master every time you edit it.",
       :type    => :file,
     },
+    :hiera_advanced_parameter_bindings => {
```

Comment:
Suggest naming this `:data_binding_lookup_options`, since the mechanism is not a hiera specific. It works with other data binding types as well.